app/services/albums.js

```diff
@@ -9,3 +12,27 @@ exports.getAlbums = () =>
     uri: `${ALBUMS_URL}${ALBUMS_ENDPOINT}`,
     json: true
   });
+
+exports.findOrBuy = (userId, albumId) => {
+  const alreadyBought = {
+    where: {
+      user_id: userId,
+      album_id: albumId
+    }
+  };
+
+  return Purchase.findAll(alreadyBought)
+    .then(purchases => {
+      const payload = { value: null, error: null };
+      if (purchases.length > 0) {
+        throw new Error(`Album ${albumId} has already being bought by User ${userId}`);
+      }
+      return Purchase.create({ userId, albumId });
+    })
+    .then(purchase => {
+      return User.findById(userId)
+        .then(user => user.addPurchases([purchase]))
+        .then(() => purchase);
+    })
+    .then(purchase => purchase.reload());
```

Comment:
why reload?

Reply:
because the reference to the user is not loaded when I add it to the user itself, so I need to reload the purchase object for this.